### ISSUE TITLE
Documentation needed

### ISSUE THREAD

**Issue by parweb** [westerndigitalcorporation/blb#1]
It would be cool to have some documentation to test the repo in multi server mode

**Comment by brendanodwyer** [comment]
Yeah it would be nice to see some more documentation related to how some of the components fit together and how the project is designed to run in multi-server mode. Also any settings or guidelines for running this(non-production) just to try out would be helpful.

**Comment by dnr** [comment]
Thanks for your interest. More docs would certainly be useful, and I'll try to migrate some of our internal docs in the coming weeks.

It would be helpful to know what sort of environments you're interested in testing things out in. E.g., machines in a rack with ssh access, VMs on EC2, containers in Mesos, etc.?

In any case, you'll probably end up modifying the scripts in [scripts](https://github.com/westerndigitalcorporation/blb/tree/master/scripts) for your environment and distributing them with your existing deployment tools. You might need to add a `-masters` flag ([spec](https://github.com/westerndigitalcorporation/blb/blob/master/internal/blbrpc/master_failover_connection.go#L37)) if you can't make the dns name `blb-master` resolve to something useful.

As far as settings, the defaults should work fine for a test cluster. Most of the tunables can be tuned with config files: use the `-masterCfg`, `-curatorCfg`, and `-tractserverCfg` flags to point to JSON files. The structs are here: [master](https://github.com/westerndigitalcorporation/blb/blob/master/internal/master/config.go), [curator](https://github.com/westerndigitalcorporation/blb/blob/master/internal/curator/config.go), [tractserver](https://github.com/westerndigitalcorporation/blb/blob/master/internal/tractserver/config.go). For master and curator, you can tune the raft parameters with a `-raftCfg` file also.

**Comment by brendanodwyer** [comment]
Honestly i wanted to test it out on Bare Metal, but also inside of containers. I have seen some presentations and documentation that came from Bloomberg earlier this year about running Hadoop on top of Kubernetes, and i was wondering if perhaps this could run on top of Kubernetes as well.

I was also curious if you guys had broken out list of things that you wanted to finish and build out with this, because the system as a whole seems very interesting and i wouldn't mind helping to take a crack at some of it.
